### ISSUE TITLE
[BUG] Protocol error (Network.getResponseBody): Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED)

### ISSUE THREAD

**Issue by selfisekai** [microsoft/playwright-python#738]
**Context:**

- Playwright Version: [what Playwright version do you use?] 1.11.2
- Operating System: [e.g. Windows, Linux or Mac] Linux (amd64), macOS Big Sur 11.4 (arm64)
- Python Version: [e.g. 3.8, 3.9] 3.9.4 (Linux), 3.9.5 (macOS)

**Code Snippet**

Can't reproduce in a short snippet, but the error appears for me like every 10-15 times on my code: https://git.sakamoto.pl/laudompat/haruhi-dl.git
```sh
$ python3 -m haruhi_dl https://www.tiktok.com/@allthingscelebrity -v --no-headless-playwright
```
(browser can be chosen for testing with a CLI option like this: `--force-playwright-browser=chromium`)

**Describe the bug**

Seems to only be applicable to firefox, and not chromium or webkit.

Results in this traceback:
```
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/laura/haruhi-dl/haruhi_dl/__main__.py", line 22, in <module>
    haruhi_dl.main()
  File "/home/laura/haruhi-dl/haruhi_dl/__init__.py", line 485, in main
    _real_main(argv)
  File "/home/laura/haruhi-dl/haruhi_dl/__init__.py", line 475, in _real_main
    retcode = hdl.download(all_urls)
  File "/home/laura/haruhi-dl/haruhi_dl/HaruhiDL.py", line 2123, in download
    res = self.extract_info(
  File "/home/laura/haruhi-dl/haruhi_dl/HaruhiDL.py", line 820, in extract_info
    return self.__extract_info(url, ie, download, extra_info, process)
  File "/home/laura/haruhi-dl/haruhi_dl/HaruhiDL.py", line 827, in wrapper
    return func(self, *args, **kwargs)
  File "/home/laura/haruhi-dl/haruhi_dl/HaruhiDL.py", line 848, in __extract_info
    ie_result = ie.extract(url)
  File "/home/laura/haruhi-dl/haruhi_dl/extractor/common.py", line 549, in extract
    ie_result = self._real_extract(url)
  File "/home/laura/haruhi-dl/haruhi_dl/extractor/tiktok.py", line 227, in _real_extract
    items = self._scroll_the_page(page, item_list_re, display_id)
  File "/home/laura/haruhi-dl/haruhi_dl/extractor/tiktok.py", line 179, in _scroll_the_page
    item_list = item_list_res.value.json()
  File "/home/laura/.local/lib/python3.9/site-packages/playwright/sync_api/_generated.py", line 433, in json
    self._sync("response.json", self._impl_obj.json())
  File "/home/laura/.local/lib/python3.9/site-packages/playwright/_impl/_sync_base.py", line 103, in _sync
    return task.result()
  File "/home/laura/.local/lib/python3.9/site-packages/playwright/_impl/_network.py", line 264, in json
    return json.loads(await self.text())
  File "/home/laura/.local/lib/python3.9/site-packages/playwright/_impl/_network.py", line 260, in text
    content = await self.body()
  File "/home/laura/.local/lib/python3.9/site-packages/playwright/_impl/_network.py", line 256, in body
    binary = await self._channel.send("body")
  File "/home/laura/.local/lib/python3.9/site-packages/playwright/_impl/_connection.py", line 36, in send
    return await self.inner_send(method, params, False)
  File "/home/laura/.local/lib/python3.9/site-packages/playwright/_impl/_connection.py", line 54, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.Error: Protocol error (Network.getResponseBody): Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsIStreamListener.onDataAvailable] convertString@chrome://juggler/content/NetworkObserver.js:882:13
getBase64EncodedResponse@chrome://juggler/content/NetworkObserver.js:841:18
getResponseBody@chrome://juggler/content/NetworkObserver.js:91:34
Network.getResponseBody@chrome://juggler/content/protocol/PageHandler.js:243:30
dispatch@chrome://juggler/content/protocol/Dispatcher.js:129:39
_dispatch@chrome://juggler/content/protocol/Dispatcher.js:69:36
receiveMessage@jar:file:///home/laura/.cache/ms-playwright/firefox-1250/firefox/omni.ja!/components/juggler.js:89:18
```

The first part of my code in the traceback can be seen here: https://git.sakamoto.pl/laudompat/haruhi-dl/-/blob/9373a2f66732283d02340809f95f18337d92edb9/haruhi_dl/extractor/tiktok.py#L179

Sometimes this fragment is also appended to the end of the traceback above:
```
events.js:291
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:94:16)
Emitted 'error' event on Socket instance at:
    at errorOrDestroy (internal/streams/destroy.js:108:12)
    at onwriteError (_stream_writable.js:424:5)
    at onwrite (_stream_writable.js:445:5)
    at internal/streams/destroy.js:50:7
    at Socket.dummyDestroy [as _destroy] (internal/bootstrap/switches/is_main_thread.js:97:3)
    at Socket.destroy (internal/streams/destroy.js:38:8)
    at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:95:12) {
  errno: 'EPIPE',
  code: 'EPIPE',
  syscall: 'write'
}
```

**Comment by mxschmitt** [comment]
I tried to reproduce it, but was not able. How often can you reproduce it? Maybe its fixed in 1.12.1.

**Comment by mxschmitt** [comment]
Closed for now, please reopen if there is a minimum reproducible available.
